include/alpaka/kernel/TaskKernelGenericSycl.hpp

```diff
@@ -136,32 +231,80 @@ namespace alpaka
 
             auto output_stream = sycl::stream{buf_size, buf_per_work_item, cgh};
 #    endif
-            cgh.parallel_for<detail::kernel<TAcc, TKernelFnObj, TArgs...>>(
-                sycl::nd_range<TDim::value>{global_size, local_size},
-                [=](sycl::nd_item<TDim::value> work_item)
-                {
-#    ifdef ALPAKA_SYCL_IOSTREAM_ENABLED
-                    auto acc = TAcc{
-                        item_elements,
-                        work_item,
-                        dyn_shared_accessor,
-                        st_shared_accessor,
-                        global_fence_dummy,
-                        local_fence_dummy,
-                        output_stream};
+       
+            constexpr std::size_t sub_group_size = trait::WarpSize<TKernelFnObj, TAcc>::warp_size;
+            bool supported = false;
+
+            if constexpr(sub_group_size == 0) {
+              // no explicit subgroup size requirement
+              LAUNCH_SYCL_KERNEL_WITH_DEFAULT_SUBGROUP_SIZE
+              supported = true;
+            } else {
```

Comment:
Isn't the warp size something that should be specified as part of the work division?

Reply:
do you mean add the warp size as a member of the work division?
The reason behind this implementation is to leave things unchanged for the other backends and also for SYCL when the warpSize is not set (i.e. is 0). That's why the warp size is an optional trait.

Reply:
Yes. Your implementation is viable too, but for me, logically, the warp size, since we can apparently choose it now and I think it relates to SIMD widths, is a division of work. But we can maybe leave it like it is for this PR.

Reply:
I agree on leaving it as it is for now, but we can discuss about it. 
The warp size (WS) must be a constant (`constexpr` or template parameter) because it will go in an attribute (i.e. `[[sycl::reqd_sub_group_size(WS)]]`). How would you add it to the work division?

Reply:
We were discussing some time back that it should also be possible to define the workdiv `constexpr`. We also have an idea on how to do it, see #1824.

Anyway, let's leave it as is.